### PR TITLE
OOP school library

### PR DESCRIPTION
**Pull request by RitobrotoMukherjee**
### Project requirements that I have done
- Created class `Person` with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise. **Changed the name from `is_of_age` **->** `of_age` to fix linters errors.**
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Created class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Created class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- Each class is saved in respective separate files.
- Added an extra file `index.rb` to test all the classes and their methods.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    false
+  end
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
```
Comment (muneebulrehman) [inline person.rb]: - Optional 
Any function in ruby returns the last line automatically. Here you don't need to explicitly mention the return statement.
Instead, you can do it as below.

def of_age?
    @age >= 18
end

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1...99_999)
+    @name = name
+    @age = age
+    @permission = parent_permission
+  end
+
+  def can_use_services?
+    return true if @permission and of_age?
```
Comment (muneebulrehman) [inline person.rb]: It should return true if either of these conditions is true. I would recommend replacing `and` by `or`.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @RitobrotoMukherjee , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Good PR description
- Great documentation

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (RitobrotoMukherjee) [reply person.rb]: Oh, awesome!! I didn't think of this. Thanks for the tips.

File: person.rb (hunk 2 above)
Comment (RitobrotoMukherjee) [reply person.rb]: Oops! I thought both should be true. fixing right away. Thanks

**Review by elmar8287 — APPROVED** [review]
Hi, @RitobrotoMukherjee.

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 

### Highlights

- All suggestions are implemented 👍 

### Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**